src/hooks/use-registration-data.ts

```diff
@@ -0,0 +1,18 @@
+import { useQuery } from "preact-fetching";
+import { useContext } from "preact/hooks";
+import { ReportFilterContext } from "../contexts/report-filter-context";
+import { loadData } from "../models/daily-registrations-report-data";
+
+interface RegistrationDataOptions {
+  start: Date;
+  finish: Date;
+}
+
+function useRegistrationData({ start, finish }: RegistrationDataOptions) {
+  const { env } = useContext(ReportFilterContext);
+  const { data } = useQuery(`daily-registrations-${finish.valueOf()}`, () => loadData(finish, env));
+
+  return data?.filter((entry) => entry.date >= start && entry.date <= finish);
```

Comment:
@zachmargolis Wondering if this might explain the performance issues noticed in #87 ? I noticed that we were feeding the full report's set of data to `Plot`, which dates back several years.

Reply:
ummm that definitely checks out

Reply:
so looking at this, this has the side effect of limiting the data rendered in the Daily Registrations Report (previously that report used to show all-time in the graph, but limit the data in the table). Was that an intentional change here?

Reply:
> so looking at this, this has the side effect of limiting the data rendered in the Daily Registrations Report (previously that report used to show all-time in the graph, but limit the data in the table). Was that an intentional change here?

It wasn't intentional, no. I might just make the date range optional so then the daily registrations report can still receive the full data set.

Reply:
> > so looking at this, this has the side effect of limiting the data rendered in the Daily Registrations Report (previously that report used to show all-time in the graph, but limit the data in the table). Was that an intentional change here?
> 
> It wasn't intentional, no. I might just make the date range optional so then the daily registrations report can still receive the full data set.

Fixed in afb19f5.